lib/rubocop/cop/samesystem/delegate_private.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Samesystem
+      # @example DelegatePrivate
+      #   # bad
+      #   class Foo
+      #     def bar=Bar.new
+      #
+      #     private
+      #
+      #     delegate :baz, to: :bar
+      #   end
+      #
+      #   # bad
+      #   class Foo
+      #     delegate :baz, to: :bar, private: true
+      #
+      #     def bar=Bar.new
+      #   end
+      #
+      #   # good
+      #   class Foo
+      #     def bar=Bar.new
+      #
+      #     private
+      #
+      #     delegate :baz, to: :bar, private: true
+      #   end
+      class DelegatePrivate < Cop
+        def on_send(node)
+          mark_scope(node)
+          return unless delegate_node?(node)
+
+          if private_scope?(node) && !private_delegate?(node)
```

Comment:
Looks like guard opportunity
```
return unless private_scope?(node)
```

Reply:
not really. `elsif public_scope?(node) && private_delegate?(node)` would be skipped with such guard